src/controllers/controllerengine.cpp

```diff
@@ -835,13 +835,25 @@ void ControllerEngineConnectionScriptValue::disconnect() {
    Input:   the ControllerEngineConnection to trigger
    -------- ------------------------------------------------------ */
 void ControllerEngine::triggerControl(const ControllerEngineConnection conn) {
-    ControlObjectScript* coScript = getControlObjectScript(conn.key.group, conn.key.item);
+    if (m_pEngine == nullptr) {
+        return;
+    }
 
-    if (m_pEngine == nullptr || coScript == nullptr) {
+    ControlObjectScript* coScript = getControlObjectScript(conn.key.group, conn.key.item);
+    if (coScript == nullptr) {
         return;
     }
 
-    coScript->emitValueChanged();
+    QScriptValueList args;
```

Comment:
I think there's quite a bit of work to do on these classes and I don't really want to go down that rabbit hole for 2.1. I don't understand why ControllerEngineConnection is even a class. It seems like it should be a struct of ControllerEngineConnectionScriptValue.